CoffeeMachineController.cpp

```diff
@@ -169,6 +176,69 @@ class CoffeeMachineController
     response.send(Http::Code::Ok, res.dump(4));
   }
 
+  void LedStripState(const Rest::Request &request, Http::ResponseWriter response)
+  {
+    json res;
+    // We can see how dirty the coffee machine is before cleaning it
+    string color;
+    bool state = coffeeMachine.getLedStripState();
+    if (state == false)
+    {
+      res["status"] = "LedStrip is off";
+    }
+    else 
+    {
+      
+      color=coffeeMachine.getLedStripColor();
+      res["status"]="LedStrip is on with color"+color;
+      
+    }
+    
+
+    //need to add this everytime
+    response.headers().add<Pistache::Http::Header::ContentType>(MIME(Application, Json));
+    //send back json response
+    response.send(Http::Code::Ok, res.dump(4));
+  }
+
+
+  void LedStripSet(const Rest::Request &request, Http::ResponseWriter response)
+  {
+    // Need to explicitly use json::parse (not just = request.body()) or else it won't work :/
+    json req = json::parse(request.body());
+    cout << req.dump(4); //4 spaces as tab in json
+
+    string color=req["color"];
+    string state = req["state"];
+
+    json res;
+    
+    
+    if (regex_match (state, regex("[0-1]") ) && regex_match (color, regex("#[a-zA-Z0-9]{6}") ))
```

Comment:
valorile hex contin litere de la 'a' la 'f', nu pana la 'z'

Reply:
thx,that's true